ha/core/system_health/system_health.py

```diff
@@ -414,17 +414,19 @@ def process_event(self, healthevent: HealthEvent):
                     and healthevent.source == HEALTH_EVENT_SOURCES.MONITOR.value:
                     # If health is already stored and its a node_health, check further
                     stored_genration_id = current_health_dict["events"][0]["specific_info"]["generation_id"]
+                    stored_status = current_health_dict["events"][0]["status"]
                     incoming_generation_id = healthevent.specific_info["generation_id"]
                     incoming_health_status = healthevent.event_type
                     pod_restart_val = current_health_dict["events"][0]["specific_info"]["pod_restart"]
                     # Update the current health value itself.
                     latest_health = EntityHealth.read(current_health)
                     if stored_genration_id and (stored_genration_id != incoming_generation_id):
-                        if incoming_health_status == HEALTH_EVENTS.ONLINE.value:
+                        if (incoming_health_status == HEALTH_EVENTS.ONLINE.value) and (stored_status != HEALTH_EVENTS.FAILED.value):
```

Comment:
This will solve the problem when pod is deleted using replicaset.
But what if pod is deleted using kubectl delete where old pod statsus is starting or offline?

Reply:
Offline constant value currently not in HEATLH_EVENT class so for offline condition will add during ticket : https://jts.seagate.com/browse/CORTX-30061

Reply:
@Madhura-08  if pod is "starting" it means that its "online" from monitor perspective; so we will need to check for restart in that case.
For "offline" that state will get added in CORTX-30061 and a TODO needs to be added here.
Also the DoD of CORTX-30061 needs to be updated to mention that this check here needs to be enhanced
@gauri-bhosale  as discussed already hope you have updated the DoD and adding a TODO here

Reply:
If required you can add that in the constant. But, I think its not needed. 
For replicaset, I gone though the code. But, I don't see any path where online event will be captured.
1. First if is to check generation_id is different or not -> Yes in case of replicaset
2. Check added by Gauri -> False
3. elif pod_restart_val -> False
After, no execution. 
@gauri-bhosale Can you please recheck?
cc @ArchanaLimaye

Reply:
Yes @Madhura-08 . Either this check needs to be on line 423 OR an additional else is needed for this .
Gauri, please  address this and also the TODO

Reply:
yes @Madhura-08  above steps are correct.

Reply:
Done, and tested well,  please check new changes.

Reply:
Ok..This looks good now. Thanks!